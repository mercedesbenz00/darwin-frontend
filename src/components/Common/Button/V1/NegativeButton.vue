<template>
  <button-template
    v-bind="$attrs"
    class="negative-button"
    v-on="$listeners"
  >
    <slot>LABEL MISSING!</slot>
  </button-template>
</template>

<script lang="ts">
import { defineComponent } from 'vue'

import ButtonTemplate from './ButtonTemplate.vue'

export default defineComponent({
  name: 'NegativeButton',
  components: { ButtonTemplate }
})
</script>

<style lang="scss" scoped>
.negative-button {
  @include row--center;

  border: 1px solid $colorCrimson;
  font-family: $fontFamilyMulish;
  font-weight: bold;

  color: $colorCrimson;
  background-color: rgba($colorCrimsonLight, 0.2);

  &:hover {
    background-color: $colorCrimson;
    color: $colorWhite;
  }

  &:active {
    background-color: rgba($colorCrimson, 0.8);
  }
}
</style>
