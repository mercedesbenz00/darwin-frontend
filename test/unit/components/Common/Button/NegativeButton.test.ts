import { createLocalVue, shallowMount, Stubs } from '@vue/test-utils'

import NegativeButton from '@/components/Common/Button/V1/NegativeButton.vue'

const localVue = createLocalVue()
const stubs: Stubs = {
  ButtonTemplate: {
    template: '<button><slot></slot></button>'
  }
}
const slots = {
  default: 'A message'
}

it('matches snapshot', () => {
  const wrapper = shallowMount(NegativeButton, { localVue, slots, stubs })
  expect(wrapper).toMatchSnapshot()
})

it('renders slot', () => {
  const wrapper = shallowMount(NegativeButton, { localVue, slots, stubs })
  expect(wrapper.text()).toContain(slots.default)
})
