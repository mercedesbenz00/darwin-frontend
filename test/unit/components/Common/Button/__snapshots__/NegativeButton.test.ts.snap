// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `<button class="negative-button">A message</button>`;
